.circleci/config.yml

```diff
@@ -1,70 +1,168 @@
-            golang:1 go test -timeout 20m -v -tags k8srequired ./${E2E_TEST_DIR}/
-
-    - run:
-        name: Debug
-        command: |
-          export KUBECONFIG=$(./e2ectl kubeconfig path)
-          ./kubectl describe rs -n kube-system
-          ./kubectl describe rs -n giantswarm
-          ./kubectl describe podsecuritypolicies
-        when: always
-
-    - run:
-        name: Export logs
-        command: |
-          ./e2ectl logs export --logs-dir=kind/logs
-        when: always
-
-    - store_artifacts:
-        path: kind/logs
-        destination: logs
-
-    - run:
-        name: Finish with cleanup, no matter if the test succeeded or not
-        command: ./e2ectl cluster delete
-        when: always
-
 version: 2.1
 orbs:
   architect: giantswarm/architect@0.4.5
 
+  inline-orb:
+    executors:
+      machine-executor:
+        machine: true
+    commands:
+      install-go:
+        steps:
+          - run:
+              name: "Remove old Go"
+              command: |
+                sudo rm -rf /usr/local/go
+          - run:
+              name: "Download Go"
+              command: |
+                wget https://dl.google.com/go/go1.13.4.linux-amd64.tar.gz
+          - run:
+              name: "Check downloaded Go checksum"
+              command: |
+                [[ "$(sha256sum go1.13.4.linux-amd64.tar.gz | cut -d ' ' -f 1)" == "692d17071736f74be04a72a06dab9cac1cd759377bd85316e52b2227604c004c" ]]
+          - run:
+              name: "Install Go"
+              command: |
+                sudo tar -C /usr/local -xzf go1.13.4.linux-amd64.tar.gz
+          - run:
+              name: "Set Go environment"
+              command: |
+                go env -w GOBIN=/home/circleci/.go_workspace/bin
+          - run:
+              name: "Create $GOPATH directory"
+              command: |
+                sudo mkdir $(go env GOPATH) && sudo chown $USER $(go env GOPATH)
+          - run:
+              name: "Remove downloaded Go files"
+              command: |
+                rm go1.13.4.linux-amd64.tar.gz
+          - run: 
+              name: "Check Go version"
+              command: |
+                go version
+
+      install-tools:
+        steps:
+          - run:
+              name: "Install kind"
+              command: |
+                CGO_ENABLED=0 GO111MODULE="on" go get sigs.k8s.io/kind@v0.7.0
+          - run:
+              name: "Install kubectl"
+              command: |
+                curl -LO https://storage.googleapis.com/kubernetes-release/release/${KUBERNETES_VERSION}/bin/linux/amd64/kubectl
+                chmod +x ./kubectl
+                sudo mv ./kubectl /usr/local/bin/kubectl
+                kubectl version --client
+
+      create-cluster:
+        parameters:
+          kind-config:
+            type: string
+        steps:
+          - run:
+              name: "Create cluster"
+              command: |
+                cd ~/project
+                
```

Comment:
We can pass a kind config file to the test. Otherwise a minimal config is generated.